pkg/apis/eventing/v1alpha1/trigger_types.go

```diff
@@ -76,6 +85,26 @@ type TriggerFilterSourceAndType struct {
 	Source string `json:"source,omitempty"`
 }
 
+// TriggerFilterCEL filters events by evaluating the expression with the Common
+// Expression Language runtime. An event passes the filter if the expression
+// evaluates to true.
+type TriggerFilterCEL struct {
+	// Expression is the CEL expression to evaluate. Required.
+	Expression string `json:"expression,omitempty"`
+	// ParseExtensions enables parsing of dynamic extensions attached to the event
+	// and makes the extensions available in the CEL environment. If extensions
+	// cannot be parsed they will be ignored. Defaults to false.
+	//
+	// +optional
+	ParseExtensions bool `json:"parseExtensions"`
```

Comment:
I didn't add this for the boolean fields because I thought there might be subtleties with omitted values. After some more research, I think adding `omitempty` is fine as long as the default value for omitted `ParseData` and `ParseExtensions` fields is always `false`.